panic when lab ci view loses connectivity to gitlab server

Connectivity was lost to the gitlab server while tracing a build job through `lab ci view`. `lab` threw a panic and left them terminal in an inconsistent state requiring `reset` to fix the terminal.

Ideally `lab` would handle the failure and attempt to reconnect, but at minimum I would hope that it exits gracefully restoring the terminal rather than panicking.

The stack trace below corresponds to the following line of code: https://github.com/zaquestion/lab/blob/bb7113ba86face26b8a7f78b08b99b8a1bef3c15/cmd/ci_trace.go#L96

I hope the stacktrace is readable enough, I had to reformat it as the panic was displayed over the corrupted display of the `lab ci view` output that triggered the panic.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4e7f19]
                                                                                              
goroutine 69 [running]:
io/ioutil.readAll.func1(0xc0004cddb0)
    /usr/lib/go-1.11/src/io/ioutil/ioutil.go:30 +0x105
panic(0x972b40, 0xe3d1c0)
    /usr/lib/go-1.11/src/runtime/panic.go:513 +0x1b9
bytes.(*Buffer).ReadFrom(0xc0003d6070, 0x0, 0x0, 0x0, 0x0, 0 x0)
    /usr/lib/go-1.11/src/bytes/buffer.go:206 +0x89                                        
io/ioutil.readAll(0x0, 0x0, 0x200, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/lib/go-1.11/src/io/ioutil/ioutil.go:36 +0xb5                     
io/ioutil.ReadAll(0x0, 0x0, 0xc0001e6046, 0xf, 0xc000310660, 0x9, 0x0)
    /usr/lib/go-1.11/src/io/ioutil/ioutil.go:45 +0x3e
github.com/zaquestion/lab/cmd.doTrace(0xaadb60, 0xc0000220e8, 0x7f91c09ad620, 0xc00 01f20c0, 0x93bae0, 0xc000334110, 0xc0001e6046, 0xf, 0xc000310660, 0x9, ...)
    /go/src/github.com/zaquestion/lab/cmd/ci_trace.go:96 +0x31d
github.com/zaquestion/lab/cmd.jobsView.func1.1(0xc000193790, 0xc000118380)
    /go/src/github.com/zaquestion/lab/cmd/ci_view.go:326 +0x204
created by github.com/zaquestion/lab/cmd.jobsView.func1
    /go/src/github.com/zaquestion/lab/cmd/ci_view.go:325 +0x117e
```

--- a/cmd/ci_run.go
+++ b/cmd/ci_run.go
@@ -38,7 +38,7 @@ lab ci create -p engineering/integration_tests master`,
 		if err != nil {
 			log.Fatal(err)
 		}
-		fmt.Printf("%s/pipelines/%d", project.WebURL, pipeline.ID)
+		fmt.Printf("%s/pipelines/%d\n", project.WebURL, pipeline.ID)
 	},
 }
 
@@ -82,7 +82,7 @@ lab ci trigger -p engineering/integration_tests -v foo=bar master`,
 		if err != nil {
 			log.Fatal(err)
 		}
-		fmt.Printf("%s/pipelines/%d", project.WebURL, pipeline.ID)
+		fmt.Printf("%s/pipelines/%d\n", project.WebURL, pipeline.ID)
 	},
 }
 

--- a/cmd/ci_trace.go
+++ b/cmd/ci_trace.go
@@ -1,7 +1,6 @@
 package cmd
 
 import (
-	"bytes"
 	"context"
 	"fmt"
 	"io"
@@ -75,7 +74,7 @@ func doTrace(ctx context.Context, w io.Writer, pid interface{}, branch, name str
 			break
 		}
 		trace, job, err := lab.CITrace(pid, branch, name)
-		if job == nil {
+		if err != nil || job == nil || trace == nil {
 			return errors.Wrap(err, "failed to find job")
 		}
 		switch job.Status {
@@ -92,20 +91,13 @@ func doTrace(ctx context.Context, w io.Writer, pid interface{}, branch, name str
 			}
 			fmt.Fprintf(w, "Showing logs for %s job #%d\n", job.Name, job.ID)
 		})
-		// TODO: can trace be passed directly to the readseaker?
-		buf, err := ioutil.ReadAll(trace)
+		_, err = io.CopyN(ioutil.Discard, trace, offset)
+		lenT, err := io.Copy(os.Stdout, trace)
 		if err != nil {
 			return err
 		}
-		r := bytes.NewReader(buf)
-		r.Seek(offset, io.SeekStart)
-		new, err := ioutil.ReadAll(r)
-		if err != nil {
-			return err
-		}
+		offset += int64(lenT)
 
-		offset += int64(len(new))
-		fmt.Fprint(w, string(new))
 		if job.Status == "success" ||
 			job.Status == "failed" ||
 			job.Status == "cancelled" {

